### PR TITLE
change '/' method from scale Op to elementwise_div Op

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

 Bug fixes
### PR changes

APIs
### Describe

* Fix the bug of div operation result using scale method do not exactly equal the result of elementwise_div method. Remove the `_scalar_div_` function.
The bug is as follow.
image

And after fixing the bug
image

* Remove `__div__` ,`__rdiv__` methods which do not be defined in python3 standard.

### CODE REVIEW

**Review by chenwhql — COMMENTED** [review]
better to add a unittest for this case?

File: python/paddle/fluid/layers/math_op_patch.py [hunk 1]
```diff
@@ -337,12 +335,8 @@ def __impl__(self, other_var):
         #  a*b == b*a. Do not need to reverse explicitly
         ('__rmul__',
          _binary_creator_('__rmul__', 'elementwise_mul', False, _scalar_mul_)),
-        ('__div__', _binary_creator_('__div__', 'elementwise_div', False,
-                                     _scalar_div_)),
         ('__truediv__', _binary_creator_('__truediv__', 'elementwise_div',
-                                         False, _scalar_div_)),
-        ('__rdiv__', _binary_creator_('__rdiv__', 'elementwise_div', True,
```
Comment (zhwesky2010) [inline python/paddle/fluid/layers/math_op_patch.py:344]: \_\_rdiv\_\_ 在python3也没有了吗

Comment (MingMingShangTian) [reply python/paddle/fluid/layers/math_op_patch.py:344]: 是的，没有了。 参考python3 的文档 https://docs.python.org/3/reference/datamodel.html

**Review by zhwesky2010 — DISMISSED** [review]

File: python/paddle/fluid/tests/unittests/test_tensor_scalar_type_promotion_dynamic.py [hunk 2]
```diff
@@ -187,6 +187,13 @@ def test_tensor_div_scalar(self):
         c = paddle.full([2, 2, 2], 0.5, dtype="float32")
         self.check_operation(a, b, c, '/')
 
+        # tensor(float32) / scalar(int)
```
Comment (chenwhql) [inline python/paddle/fluid/tests/unittests/test_tensor_scalar_type_promotion_dynamic.py:190]: 在静态图的那个单测里面也加一下？

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**